### DISCUSSION TITLE
Amazon Cognito - without hosted UI

### DISCUSSION THREAD

**Discussion by scarsam** [nextauthjs/next-auth#1092] [Help]
👋 I'm planning on testing out [Amazon Cognito Provider](https://next-auth.js.org/providers/cognito). Does anyone know if it requires me to use their hosted UI to use this together with `next-auth`?

I would like to save the users through some other provider (email only or social login) with my own UI but have the user show up in AWS user-pool. This to make it easier to authorize DB calls inside AWS.

**Comment by thetaPC** [comment]
This is possible. I've been able to have users login by verifying their auth details with Cognito. This gives me the control on what the form looks like. I would recommend [one of these libraries](https://www.maxivanov.io/aws-cognito-amplify-vs-amazon-cognito-identity-js-vs-aws-sdk/) to set it up.

**Reply by aswwwanth** [reply]
I am not quite sure if those libraries can be used with `next-auth`. As far as I understand if I use one of these libraries then I will have to take care of session management in the Next.js app, but instead I want `next-auth` to take care of session management for me. I am relatively new to Next.js, next-auth and AWS Cognito so please correct me if I am wrong 🙂

**Comment by balazsorban44** [comment]
There is a relevant issue on this, see #2525.
